kodilla-stream/src/main/java/com/kodilla/stream/world/World.java

```diff
@@ -0,0 +1,20 @@
+package com.kodilla.stream.world;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class World {
+    private final List<Continent> world = new ArrayList<>();
+
+    public void addContinent(Continent continent) {
+        world.add(continent);
+    }
+
+    public BigDecimal getPeopleQuantity() {
+        return world.stream()
+                .flatMap(continent -> continent.getContinent().stream())
+                .map(Country::getPeopleQuantity)
+                .reduce(BigDecimal.ZERO, (sum, current) -> sum = sum.add(current));
```

Comment:
Przypisanie `sum =` nie jest potrzebne. Ono zmienia argument lambdy, czyli nie ma żadnego znaczenia. Kod działa poprawnie dlatego, że wyrażenie `x = y` ma wartość `y`.
Ergo - wystarczy `(sum, current) -> sum.add(current)`. Liczy się wartość zwracana z lambdy, a nie przypisanie.
Jeszcze ładniej napiszesz to w postaci referencji `BigDecimal::add`.